agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/HttpClientTracer.java

```diff
@@ -0,0 +1,179 @@
+
+    if (clientSpan != null) {
+      // We don't want to create two client spans for a given client call, suppress inner spans.
+      return DefaultSpan.getInvalid();
+    }
+
+    Span current = TracingContextUtils.getSpan(context);
+    Span span = tracer.spanBuilder(name).setSpanKind(Kind.CLIENT).setParent(current).startSpan();
+    onRequest(span, request);
+    return span;
+  }
+
+  private Span onRequest(final Span span, final REQUEST request) {
+    assert span != null;
+    if (request != null) {
+      span.setAttribute(SemanticAttributes.HTTP_METHOD.key(), method(request));
+
+      String userAgent = requestHeader(request, USER_AGENT);
+      if (userAgent != null) {
+        SemanticAttributes.HTTP_USER_AGENT.set(span, userAgent);
+      }
+
+      // Copy of HttpServerDecorator url handling
+      try {
+        URI url = url(request);
+        if (url != null) {
+          StringBuilder urlBuilder = new StringBuilder();
+          if (url.getScheme() != null) {
+            urlBuilder.append(url.getScheme());
+            urlBuilder.append("://");
+          }
+          if (url.getHost() != null) {
+            urlBuilder.append(url.getHost());
+            setPeer(span, url.getHost(), null);
+            if (url.getPort() > 0) {
+              span.setAttribute(SemanticAttributes.NET_PEER_PORT.key(), url.getPort());
+              if (url.getPort() != 80 && url.getPort() != 443) {
+                urlBuilder.append(":");
+                urlBuilder.append(url.getPort());
+              }
+            }
+          }
+          String path = url.getPath();
+          if (path.isEmpty()) {
+            urlBuilder.append("/");
+          } else {
+            urlBuilder.append(path);
+          }
+          String query = url.getQuery();
+          if (query != null) {
+            urlBuilder.append("?").append(query);
+          }
+          String fragment = url.getFragment();
+          if (fragment != null) {
+            urlBuilder.append("#").append(fragment);
+          }
+
+          span.setAttribute(SemanticAttributes.HTTP_URL.key(), urlBuilder.toString());
+
+          if (Config.get().isHttpClientTagQueryString()) {
+            span.setAttribute(MoreAttributes.HTTP_QUERY, query);
+            span.setAttribute(MoreAttributes.HTTP_FRAGMENT, fragment);
+          }
+        }
+      } catch (final Exception e) {
+        log.debug("Error tagging url", e);
+      }
+    }
+    return span;
+  }
+
+  private Span onResponse(final Span span, final RESPONSE response) {
+    assert span != null;
+    if (response != null) {
+      Integer status = status(response);
+      System.out.println("status: " + status);
+      if (status != null) {
+        span.setAttribute(SemanticAttributes.HTTP_STATUS_CODE.key(), status);
+        span.setStatus(HttpStatusConverter.statusFromHttpStatus(status));
+      }
+    }
+    return span;
+  }
+
+  private String spanNameForRequest(final REQUEST request) {
```

Comment:
Should this be protected? How would we instrument an HTTP library with RPC semantics, e.g., `Retrofit`?

Probably don't have to decide in this PR though.

Reply:
If needed, i will change it later.. not sure which decorator is that?